### PR TITLE
Document CSRF protection

### PR DESCRIPTION
**Pull request by rnewson**
COUCHDB-2762

### CODE REVIEW

File: src/config/http.rst [hunk 1]
```diff
@@ -507,6 +507,71 @@ Cross-Origin Resource Sharing
...
+        }
+
+    .. config:option:: secret
+
+        All CSRF cookies are signed by the server using this value. A
+        random value will be chosen if you don't specify it, but we
+        recommend setting it yourself, especially if you are running a
+        cluster of more than one node. The secret must match on all
+        nodes in a cluster to avoid sadness.
+
```
Comment (kxepal) [inline src/config/http.rst]: Would be good to have an example here as other options have.

File: src/config/http.rst [hunk 2]
```diff
@@ -507,6 +507,71 @@ Cross-Origin Resource Sharing
...
+    .. config:option:: timeout
+
+        All CSRF cookies expire after `timeout` seconds. The default
+        is an hour.
+
+
```
Comment (kxepal) [inline src/config/http.rst]: `make check` isn't happy because of addition newline here. Trying to keep same style all around.

**Comment by kxepal** [comment]
Great addition! LGFM except examples and failed linter check.

**Comment by rnewson** [comment]
the toolchain here was huge so I haven't even built it.

**Comment by kxepal** [comment]
It's actually strange that it's all is required even you need just make html version. I'll check makefile rules.  But anyway, that's why travis here to help (:

**Comment by rnewson** [comment]
tests pass (MacTex download continues to fail, so I've given up with local attempt, but I really wish we didn't need the 2.68GB blob of LaTeX just to generate our docs).

File: src/config/http.rst [hunk 3]
```diff
@@ -507,6 +507,85 @@ Cross-Origin Resource Sharing
...
+          setRequestHeader("X-CouchDB-CSRF", "true");
+        }
+
+    .. config:option:: mandatory
+
+        CouchDB can insist on CSRF Cookie/Header for all requests
+        (except those to the welcome handler, /, so you can acquire a
+        cookie) with this setting. The default is false.
+
+            [csrf]
```
Comment (kxepal) [inline src/config/http.rst]: This wouldn't render as code block, but just as a sub paragraph. Wrap it with `.. code-block:: ini`.

**Comment by rnewson** [comment]
I updated it to match the other ways we do it in other sections. Emacs indicated I did something right but I'm flying blind until this 2.68gb thing arrives

**Comment by kxepal** [comment]
All good. +1

**Comment by rnewson** [comment]
no idea why it won't mark this as merged but it is merged.

**Comment by kxepal** [comment]
Hashes seems different: 13e9832 on master vs f88582f here. Could you just close it then?

**Pull request closed without merging**